src/neo-vm/ExecutionEngine.cs

```diff
@@ -443,10 +443,10 @@ private bool ExecuteInstruction()
                         }
                     case OpCode.CAT:
                         {
-                            if (!(context.EvaluationStack.Pop() is PrimitiveType item_x2))
+                            if (!context.EvaluationStack.TryPop<PrimitiveType>(out var item_x2))
                                 return false;
                             ReadOnlyMemory<byte> x2 = item_x2.ToMemory();
-                            if (!(context.EvaluationStack.Pop() is PrimitiveType item_x1))
+                            if (!context.EvaluationStack.TryPop<PrimitiveType>(out var item_x1))
```

Comment:
We should reverse each of these if statements so there's a single return false at the end of each case statement instead of after each TryPop call

Reply:
This could be done in other PR, but I am curious, this have any speed impact? or is just for readability?

Reply:
readability